solvation_analysis/tests/test_solvation_analysis.py

```diff
@@ -77,28 +85,76 @@ def u_real_named(u_real):
 
 @pytest.fixture
 def atom_groups(u_real):
-    li_atoms = u_real.atoms.select
-    pf6_atoms = u_real.atoms.select
-    bn_atoms = u_real.atoms.select
-    fec_atoms = u_real.atoms.select_atoms()
+    li_atoms = u_real.atoms.select_atoms("type 22")
+    pf6_atoms = u_real.atoms.select_atoms("byres type 20")
+    bn_atoms = u_real.atoms.select_atoms("byres type 5")
+    fec_atoms = u_real.atoms.select_atoms("byres type 21")
     atom_groups = {"li": li_atoms, "pf6": pf6_atoms, "bn": bn_atoms, "fec": fec_atoms}
-
-
-def test_solvation_analysis_imported():
-    """Sample test, will always pass so long as import statement worked"""
-    assert "solvation_analysis" in sys.modules
+    return atom_groups
 
 
 def test_get_atom_group(u_real_named):
-    return
-
-
-def test_get_closest_n_mol():
-    return
-
-
-def test_get_radial_shell():
-    return
+    u = u_real_named
+    res_group = u.residues[1:5]
+    atom_group = u.atoms[1:5]
+    res = u.residues[1]
+    atom = u.atoms[1]
+    groups = [res_group, atom_group, res, atom]
+    for group in groups:
+        assert isinstance(get_atom_group(u, group), mda.core.groups.AtomGroup)
+
+
+def test_get_closest_n_mol_grid(u_grid_1):
+    u = u_grid_1
+    test_atoms = u.atoms[[0, 5, 10, 16, 42]]
+    for test_atom in test_atoms:
+        atoms = get_closest_n_mol(u, test_atom)
+        assert len(atoms) == 6
+
+
+def test_get_closest_n_mol_real(u_real, atom_groups):
```

Comment:
I might be missing something, but AFAIK this test primarily tests the behaviour of `get_closest_n_mol` against itself.  I think this is a good test to have, but perhaps some clarification in name required.

I also think you should add a test with some hardcoded values in `u_real` like you have done in `test_get_radial_shell_grid` that actually tests if the selected molecules are the closest ones.

Reply:
Got it, added some hardcoded values.

Currently I have the testing organized by what data structure the tests are running on. Maybe I should reorganize to have each function test a specific aspect of functionality? Is there any naming convention I should follow if that's the case?

Reply:
Whatever makes the most sense to you. When we move to an analysis class, it will be easiest to group all the tests for that class together.